backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -428,6 +433,120 @@ public FutureExperimentRunDAO(
+                  .bindList("linkedDatasetVersionIds", linkedDatasetVersionIds)
+                  .map(
+                      (rs, ctx) ->
+                          new AbstractMap.SimpleEntry<>(
+                              rs.getString("commit_hash"), rs.getLong("repository_id")))
+                  .list();
+            })
+        .thenCompose(
+            datasetVersionDatasetList -> {
+              // Convert list of dataset and dataset version mapping into Map
+              Map<String, String> datasetVersionDatasetMap = new HashMap<>();
+              for (var entry : datasetVersionDatasetList) {
+                datasetVersionDatasetMap.put(entry.getKey(), String.valueOf(entry.getValue()));
+              }
+              return InternalFuture.completedInternalFuture(datasetVersionDatasetMap);
+            },
+            executor)
+        .thenCompose(
+            datasetVersionDatasetMap -> {
+              // Validate linked dataset in dataset version exists in database records if not then
+              // we will throw the error
+              Set<String> datasetIds = new HashSet<>();
+              for (Artifact dataset : datasets) {
+                String datasetVersionId = dataset.getLinkedArtifactId();
+                if (datasetVersionDatasetMap.containsKey(datasetVersionId)
+                    && datasetVersionDatasetMap.get(datasetVersionId) != null) {
+                  datasetIds.add(datasetVersionDatasetMap.get(datasetVersionId));
+                } else {
+                  return InternalFuture.failedStage(
+                      new InvalidArgumentException(
+                          "Dataset not found for dataset version: " + datasetVersionId));
+                }
+              }
+
+              // If all dataset with linked dataset versions found in database then checking
+              // authorization for the current user for those datasets
+              List<InternalFuture<String>> internalFutures = new LinkedList<>();
+              for (String datasetId : datasetIds) {
+                try {
+                  internalFutures.add(
+                      InternalFuture.completedInternalFuture(datasetId)
+                          .thenCompose(
+                              id ->
+                                  checkEntityPermissionBasedOnResourceTypes(
+                                          Collections.singletonList(datasetId),
+                                          ModelDBActionEnum.ModelDBServiceActions.READ,
+                                          ModelDBServiceResourceTypes.DATASET)
+                                      .thenCompose(
+                                          unused -> InternalFuture.completedInternalFuture(id),
+                                          executor),
+                              executor));
+
+                } catch (PermissionDeniedException ex) {
```

Comment:
The command above will never throw this exception. This is left over from when you were doing the blocking call.

Reply:
No this is not about ablocking code. i have to decide about throwing `PermissionDeniedException` or not based on `errorOut` (errorOut : Throw error while creation (true) otherwise we will keep it silent (false) at fetch time) flag so i have to catch it here.